### PR TITLE
`Paywalls`: updated variable names

### PR DESCRIPTION
**Pull request by NachoSoto**
These match the new agreed upon variables.

### Other changes:
- Improved the localized unit abbreviations
- Refactored `VariableDataProvider` so all the logic is now in `Package`
- Added new tests for `Package` variables

### CODE REVIEW

File: RevenueCatUI/Data/Variables.swift [hunk 1]
```diff
@@ -59,26 +60,19 @@ extension String {
...
-            return self.periodName(locale)
-        case "subscription_duration":
-            return self.subscriptionDuration(locale) ?? ""
-        case "intro_duration":
-            return self.introductoryOfferDuration(locale) ?? ""
+        case "sub_period": return self.periodName(locale)
+        case "sub_price_per_month": return self.localizedPricePerMonth
+        case "sub_duration": return self.subscriptionDuration(locale) ?? ""
+        case "sub_offer_duration": return self.introductoryOfferDuration(locale) ?? ""
+        case "sub_offer_price": return self.localizedIntroductoryOfferPrice ?? ""
```
Comment (NachoSoto) [inline RevenueCatUI/Data/Variables.swift:75]: New variable

File: RevenueCatUI/Data/Variables.swift [hunk 2]
```diff
@@ -59,26 +60,19 @@ extension String {
...
-        case "price_per_month": return self.localizedPricePerMonth
-        case "total_price_and_per_month":
-            if !self.isSubscription || self.isMonthly {
-                return self.localizedPrice
-            } else {
-                let unit = Localization.abbreviatedUnitLocalizedString(for: .month, locale: locale)
-                return "\(self.localizedPrice) (\(self.localizedPricePerMonth)/\(unit))"
-            }
-
+        case "price_per_period": return self.localizedPricePerPeriod(locale)
```
Comment (NachoSoto) [inline RevenueCatUI/Data/Variables.swift:68]: New variable

**Review by joshdholtz — APPROVED** [review]
LGMT 💪

**Comment by NachoSoto** [comment]
Not gonna merge this until the frontend is ready.

**Pull request merged**